src/func.d

```diff
@@ -3029,6 +3008,46 @@ extern (C++) class FuncDeclaration : Declaration
         return true; // functions can be overloaded
     }
 
+    /**********************************
+     * Decide if attributes for this function can be inferred from examining
+     * the function body.
+     * Returns:
+     *  true if can
+     */
+    private final bool canInferAttributes(Scope* sc)
+    {
+        TemplateInstance ti;
+        return (fbody && !isVirtualMethod() &&
+                        /********** this is for backwards compatibility for the moment ********/
```

Comment:
"this" meaning "the two checks above" or "the three checks below"? Or something else?